# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/shistory/nsSHistory.cpp
# Commit: 0a44c410b59b
# Full Hash: 0a44c410b59bbc54bfd80ce598562e4362961d7e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-24 09:51:27
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 20dbcfc9eacc -r 0a44c410b59b docshell/shistory/nsSHistory.cpp
--- a/docshell/shistory/nsSHistory.cpp	Thu Apr 23 21:52:44 2020 +0000
+++ b/docshell/shistory/nsSHistory.cpp	Thu Apr 23 21:52:47 2020 +0000
@@ -386,7 +386,7 @@
 
     BrowsingContext* childBC = nullptr;
     if (aBC) {
-      for (BrowsingContext* child : aBC->GetChildren()) {
+      for (BrowsingContext* child : aBC->Children()) {
         // If the SH pref is on, or we are in the parent process, update
         // canonical BC directly
         if (StaticPrefs::fission_sessionHistoryInParent() ||